analysis/src/Hint.ml

```diff
@@ -32,15 +32,21 @@ let locItemToTypeHint ~full:{file; package} locItem =
         | `Field -> fromType t))
   | _ -> None
 
-let inlay ~path ~maxLength ~debug =
-  let maxlen = try Some(int_of_string maxLength) with Failure _ -> None in
+let inlay ~path ~pos ~maxLength ~debug =
```

Comment:
Do we still need `maxLength` now that ranges are considered?
I though length is implicit in the loc that vscode sends?

Reply:
yup. `maxLength` filters hints within the range (`pos`).

Reply:
And why is it needed? I though vscode would decide what range to ask for.

Reply:
Unless we measure a perf improvement it should not be there.

Reply:
It's for these cases: https://github.com/rescript-lang/rescript-vscode/pull/453#discussion_r925964108

![](https://user-images.githubusercontent.com/16160544/180066275-0bc9735c-cd07-41e4-b9d4-7a2352149009.png)

Some type hints are long and we want to filter. I added it to the binary to not filter on `server.ts` side.

Reply:
It is more convenient to filter in bin analysis

Reply:
Oh got it. I thought was also some thing similar to range.
Let me check the documentation if it's clear what it is.

Reply:
Checked: the config description that the user sees is a bit vague.
Can you clarify the description?

Reply:
Done https://github.com/rescript-lang/rescript-vscode/pull/453/commits/4df48ae9a1df6ab5f132ae7c67d813cbbaa190a8